protocol/metrics/consumer_relayserver_client.go

```diff
@@ -0,0 +1,239 @@
+	"sync"
+	"time"
+
+	"github.com/lavanet/lava/utils"
+)
+
+type ConsumerRelayServerClient struct {
+	endPointAddress    string
+	addQueue           []UpdateMetricsRequest
+	ticker             *time.Ticker
+	lock               sync.RWMutex
+	sendID             int
+	isSendQueueRunning bool
+}
+type UpdateMetricsRequest struct {
+	RecordDate      string `json:"RecordDate"`
+	Hash            string `json:"Hash"`
+	Chain           string `json:"Chain"`
+	ApiType         string `json:"ApiType"`
+	RelaysInc       uint64 `json:"RelaysInc"`
+	CuInc           int    `json:"CuInc"`
+	LatencyToAdd    uint64 `json:"LatencyToAdd"`
+	LatencyAvgCount int    `json:"LatencyAvgCount"`
+}
+
+func NewConsumerRelayServerClient(endPointAddress string) *ConsumerRelayServerClient {
+	if endPointAddress == DisabledFlagOption {
+		utils.LavaFormatDebug("CUC: Consumer Usageserver disabled")
+		return nil
+	}
+
+	cuc := &ConsumerRelayServerClient{
+		endPointAddress: endPointAddress,
+		ticker:          time.NewTicker(30 * time.Second),
+		addQueue:        make([]UpdateMetricsRequest, 0),
+	}
+
+	go cuc.relayDataSendQueueStart()
+
+	return cuc
+}
+
+func (cuc *ConsumerRelayServerClient) relayDataSendQueueStart() {
+	if cuc == nil {
+		utils.LavaFormatWarning("CUC: Warning - CUC is nil. called from relayDataSendQueueStart", nil)
+		return
+	}
+
+	utils.LavaFormatDebug("CUC: Starting relayDataSendQueueStart loop")
+
+	for range cuc.ticker.C {
+		cuc.relayDataSendQueueTick()
+	}
+}
+
+func (cuc *ConsumerRelayServerClient) relayDataSendQueueTick() {
+	if cuc == nil {
+		utils.LavaFormatWarning("CUC: Warning - CUC is nil. called from relayDataSendQueueTick", nil)
+		return
+	}
+
+	cuc.lock.Lock()
+	defer cuc.lock.Unlock()
+
+	if !cuc.isSendQueueRunning && len(cuc.addQueue) > 0 {
+		sendQueue := cuc.addQueue
+		cuc.addQueue = make([]UpdateMetricsRequest, 0)
+		cuc.isSendQueueRunning = true
+		cuc.sendID++
+		utils.LavaFormatDebug(fmt.Sprintf("CUC: Swapped queues (sendQueue length: %d) and set isSendQueueRunning to true - CCC iter:%d.", len(sendQueue), cuc.sendID))
+
+		sendID := cuc.sendID
+		cucEndpointAddress := cuc.endPointAddress
+
+		go func() {
+			utils.LavaFormatDebug(fmt.Sprintf("CUC: Sending requests (sendQueue length: %d) - async start - CCC iter:%d.", len(sendQueue), sendID))
+			cuc.sendRelayData(sendQueue, sendID, cucEndpointAddress)
+			utils.LavaFormatDebug(fmt.Sprintf("CUC: Sending requests - async end - CCC iter:%d.", sendID))
+
+			cuc.lock.Lock()
+			cuc.isSendQueueRunning = false
+			utils.LavaFormatDebug(fmt.Sprintf("CUC: Finished sending requests and set isSendQueueRunning to false - CCC iter:%d.", cuc.sendID))
+			cuc.lock.Unlock()
+		}()
+	} else {
+		utils.LavaFormatDebug(fmt.Sprintf("CUC: Skipped this iteration - CCC iter:%d.", cuc.sendID))
+	}
+}
+
+func (cuc *ConsumerRelayServerClient) SetRelayMetrics(relayMetric *RelayMetrics) error {
+	if cuc == nil {
+		utils.LavaFormatWarning("CUC: Warning - CUC is nil. called from SetRelayMetrics", nil)
```

Comment:
this is gonna spam our logs like hell. every relay. please clean all logs from nil case. 
please clean all logs that might spam from warning / error levels ( debug you can leave but only if its not too spammy)

Reply:
ok - this is easy to remove